torch/types.py

```diff
@@ -1,7 +1,26 @@
 import torch
-from typing import Union, Sequence
+from typing import Union, Sequence, List, Tuple
+
+import builtins
 
 # Convenience aliases for common composite types that we need
 # to talk about in PyTorch
 
 _TensorOrTensors = Union[torch.Tensor, Sequence[torch.Tensor]]
+
+# In some cases, these basic types are shadowed by corresponding
+# top-level values.  The underscore variants let us refer to these
+# types.  See https://github.com/python/mypy/issues/4146 for why these
+# workarounds is necessary
+_int = builtins.int
+_float = builtins.float
+_bool = builtins.bool
+
+_dtype = torch.dtype
+_device = torch.device
+_qscheme = torch.qscheme
+_size = Union[torch.Size, List[_int], Tuple[_int, ...]]
```

Comment:
This is a little inconsistent compared with all the other underscored names above, one would expect `_size` to mean `torch.size`.

The generated `_C/__init__.pyi` now has a number of signatures with:
```
self, size: _size, *, dtype: _dtype=None, layout: _layout=strided, device: Union[_device, str, None]=None,
```
I think it would make sense to change `_device` above to `_device: Union[torch.device, str, None]` so the signatures become:

```
self, size: _size, *, dtype: _dtype=None, layout: _layout=strided, device: _device=None, ...
```
And if people need to use `device` rather than that union to annotate a device parameter, they can always use the explicit `torch.device`, the fully-qualified name should avoid `mypy/issues/4146`.

Reply:
One thing to note is that these names are all grandfathered from the old type stubs. So while I'm generally amenable to changing things up, it will greatly increase the surface area of the patch.